llama_index/program/llm_prompt_program.py

```diff
@@ -0,0 +1,40 @@
+"""LLM Prompt Program."""
+from llama_index.program.base_program import BasePydanticProgram
+from llama_index.bridge.langchain import ChatOpenAI
+from llama_index.prompts.base import Prompt
+from typing import Type, Any, Optional, Union
+from llama_index.types import Model
+from abc import abstractmethod
+
+
+class BaseLLMPromptProgram(BasePydanticProgram):
```

Comment:
curious: why do we need this layer?

Reply:
would prefer less layers in inheritance if possible.

Reply:
If this is really necessary, could use a Protocol to define the interface.

Reply:
also, not sure I like this name, feels too similar to `LLMTextCompletionProgram`

Maybe `BaseLLMFunctionProgram` ?

Reply:
hm, yeah sure! was originally trying to fit LLMTextCompletion into this somehow but it wasn't a good fit